axelor-project/src/main/java/com/axelor/apps/project/web/ProjectTemplateController.java

```diff
@@ -0,0 +1,119 @@
+import com.axelor.apps.base.db.repo.PartnerRepository;
+import com.axelor.apps.project.db.Project;
+import com.axelor.apps.project.db.ProjectTemplate;
+import com.axelor.apps.project.db.repo.ProjectTemplateRepository;
+import com.axelor.apps.project.service.ProjectService;
+import com.axelor.apps.project.service.app.AppProjectService;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.axelor.meta.schema.actions.ActionView;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import java.util.LinkedHashMap;
+
+@Singleton
+public class ProjectTemplateController {
+
+  @Inject ProjectTemplateRepository projectTemplateRepo;
+  @Inject ProjectService projectService;
+  @Inject AppProjectService appProjectService;
+
+  public void createProjectFromTemplate(ActionRequest request, ActionResponse response) {
+
+    ProjectTemplate projectTemplate = request.getContext().asType(ProjectTemplate.class);
+    AppProject appProject = appProjectService.getAppProject();
+
+    if (appProject.getGenerateProjectSequence() && !projectTemplate.getIsBusinessProject()) {
+      Project project = new Project();
+
+      projectService.createProjectFromTemplate(project, projectTemplate);
+
+      response.setCanClose(true);
+
+      response.setView(
+          ActionView.define(I18n.get("Project"))
+              .model(Project.class.getName())
+              .add("form", "project-form")
+              .add("grid", "project-grid")
+              .context("_showRecord", String.valueOf(project.getId()))
+              .map());
+    } else {
+      response.setView(
+          ActionView.define("Create project from this template")
+              .model(Wizard.class.getName())
+              .add("form", "project-template-wizard-form")
+              .param("popup", "reload")
+              .param("show-toolbar", "false")
+              .param("show-confirm", "false")
+              .param("width", "large")
+              .param("popup-save", "false")
+              .context("_projectTemplate", projectTemplate)
+              .context("_businessProject", projectTemplate.getIsBusinessProject())
+              .context(
+                  "_bothFieldsRequired",
+                  projectTemplate.getIsBusinessProject()
+                      && !appProject.getGenerateProjectSequence())
+              .map());
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public void createProjectFromWizard(ActionRequest request, ActionResponse response) {
+
+    LinkedHashMap<String, Object> projectTemplateContext =
+        (LinkedHashMap<String, Object>) request.getContext().get("_projectTemplate");
+    Integer projectTemplateId = (Integer) projectTemplateContext.get("id");
+    ProjectTemplate projectTemplate = projectTemplateRepo.find(projectTemplateId.longValue());
+
+    Project project = new Project();
+
+    if (request.getContext().get("code") != null) {
```

Comment:
Store context in new variable and get values from it , its not good to use request.getContext repeatedly